static/js/message_list_view.js

```diff
@@ -1445,4 +1445,76 @@ export class MessageListView {
             message_container.status_message = false;
         }
     }
+
+    update_sticky_recipient_headers() {
+        const rows_length = this._rows.size;
+        if (!rows_length) {
+            /* No headers are present */
+            return;
+        }
+        /* Intentionally remove sticky headers class here to make calculations simpler. */
+        $(".sticky_header").removeClass("sticky_header");
+        /* visible_top is navbar top position + height for us. */
+        const visible_top = $message_viewport.message_viewport_info().visible_top;
+        /* We need date to be properly visible on the header, so partially visible headers
+        who are about to be scrolled out of view are not acceptable. */
```

Comment:
This comment seems misplaced? The next lines seems to be unrelated.

Reply:
It should make sense now.